app/namespace.go

```diff
@@ -165,277 +167,427 @@ func ReadCACerts(ctx *cli.Context) (string, error) {
+								n, existingCerts, err := c.parseExistingCerts(ctx)
+								if err != nil {
+									return err
+								}
+								var certs caCerts
+								if ctx.String(caCertificateFingerprintFlagName) != "" {
+									certs, err = removeCertWithFingerprint(
+										existingCerts,
+										ctx.String(caCertificateFingerprintFlagName),
+									)
+									if err != nil {
+										return err
+									}
+								} else {
+									readCerts, err := readAndParseCACerts(ctx)
+									if err != nil {
+										return err
+									}
+									certs, err = removeCerts(existingCerts, readCerts)
+									if err != nil {
+										return err
+									}
+								}
+								bundle, err := certs.bundle()
+								if err != nil {
+									return err
+								}
+								if n.Spec.AcceptedClientCa == bundle {
+									return errors.New("nothing to change")
+								}
+								n.Spec.AcceptedClientCa = bundle
+								y, err := ConfirmPrompt(ctx, "removing ca certificates can cause connectivity disruption if there are any clients using certificates that cannot be verified. confirm remove?")
+								if err != nil || !y {
+									return err
+								}
+								return c.updateNamespace(ctx, n)
+							},
+						},
+						{
+							Name:    "set",
+							Aliases: []string{"s"},
+							Usage:   "Set the accepted client ca certificate",
+							Flags: []cli.Flag{
+								NamespaceFlag,
+								RequestIDFlag,
+								ResourceVersionFlag,
+								CaCertificateFlag,
+								CaCertificateFileFlag,
+							},
+							Action: func(ctx *cli.Context) error {
+								cert, err := ReadCACerts(ctx)
+								if err != nil {
+									return err
+								}
+								n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+								if err != nil {
+									return err
+								}
+								if n.Spec.AcceptedClientCa == cert {
+									return errors.New("nothing to change")
+								}
+								n.Spec.AcceptedClientCa = cert
+								return c.updateNamespace(ctx, n)
+							},
+						},
 					},
 				},
-				Action: func(ctx *cli.Context) error {
-					n, err := c.getNamespace(
-						ctx.String(NamespaceFlagName),
-					)
-					if err != nil {
-						return err
-					}
-					existingName := ctx.String("existing-name")
-					if _, exists := n.Spec.SearchAttributes[existingName]; !exists {
-						return fmt.Errorf("search attribute with name '%s' does not exist", ctx.String("existing-name"))
-					}
-					newName := ctx.String("new-name")
-					if _, exists := n.Spec.SearchAttributes[newName]; exists {
-						return fmt.Errorf("search attribute with new name '%s' already exists", ctx.String("new-name"))
-					}
-					y, err := ConfirmPrompt(ctx, "renaming search attribute may cause failures if any worker is still using the old name of the search-attributes. confirm rename?")
-					if err != nil || !y {
-						return err
-					}
-					return c.renameSearchAttribute(ctx, n, existingName, newName)
+				{
+					Name:    "certificate-filters",
```

Comment:
this can be nested within accepted-client-ca instead of being its own top-level thing. thoughts?

Reply:
I vote to keep them as is.